src/reporters/apiary-reporter.coffee

```diff
@@ -107,19 +109,40 @@ class ApiaryReporter
 
           callback()
 
-    emitter.on 'test pass', (test) =>
+    emitter.on 'test pass', (test, callback) =>
+      return callback() if @serverError == true
       data = @_transformTestToReporter test
       path = '/apis/' + @configuration['apiSuite'] + '/tests/steps?testRunId=' + @remoteId
       @_performRequest path, 'POST', data, (error, response, parsedBody) ->
         logger.error error if error
+        callback()
 
-    emitter.on 'test fail', (test) =>
+    emitter.on 'test fail', (test, callback) =>
+      return callback() if @serverError == true
       data = @_transformTestToReporter test
       path = '/apis/' + @configuration['apiSuite'] + '/tests/steps?testRunId=' + @remoteId
       @_performRequest path, 'POST', data, (error, response, parsedBody) ->
         logger.error error if error
+        callback()
+
+    emitter.on 'test error', (test, error, callback) =>
+      return callback() if @serverError == true
+      data = @_transformTestToReporter test
+      data.result = 'error'
+
+      connectionErrors = ['ECONNRESET', 'ENOTFOUND', 'ESOCKETTIMEDOUT', 'ETIMEDOUT', 'ECONNREFUSED', 'EHOSTUNREACH', 'EPIPE']
+      if connectionErrors.indexOf(error.code) > -1
```

Comment:
you know I like coffee-script, can this line be more coffee-scriptish? E.g. `if error.code in connectionErrors`

and also - connectionErrors might become a constant at the beginning of the file?